File: Sources/AndroidBackend/Kotlin/CustomNumberPicker.kt
package dev.swiftcrossui.androidbackend

import android.app.Activity
import android.widget.NumberPicker

class CustomNumberPicker(activity: Activity): NumberPicker(activity) {
    init {
        minValue = 0
    }
    
    fun update(onChange: SwiftAction, options: Array<String>, isEnabled: Boolean) {
        maxValue = options.size
        displayedValues = arrayOf("") + options
        
        setOnValueChangedListener { _, _, _ ->
            onChange.call()
        }
        
        setEnabled(isEnabled)
    }
}

File: Sources/AndroidBackend/CustomNumberPicker.swift
import SwiftJava
import AndroidKit

@JavaClass(
    "dev.swiftcrossui.androidbackend.CustomNumberPicker",
    extends: AndroidKit.NumberPicker.self
)
class CustomNumberPicker: JavaObject {
    @JavaMethod
    @_nonoverride convenience init(
        _ activity: Activity?,
        environment: JNIEnvironment? = nil
    )

    @JavaMethod
    func update(_ onChange: SwiftAction?, _ options: [String], _ isEnabled: Bool)
    
    // Inheritied from NumberPicker; only present to avoid the extra as() call Swift-side.
    @JavaMethod
    func getValue() -> Int32
}
